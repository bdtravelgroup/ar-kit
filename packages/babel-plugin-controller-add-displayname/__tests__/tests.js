const babel = require('babel-core');
const fs = require('fs');
const path = require('path');

const fixturesDir = path.join(__dirname, 'fixtures');

const pluginPath = path.join(__dirname, '../../babel-plugin-controller-add-displayname');

function transformFile(filename) {
  return babel.transformFileSync(filename, {
    presets: ['react', 'stage-1'],
    plugins: [
      [pluginPath, { knownComponents: ['Component5a', 'Component5b', 'Component5c'] }],
      'transform-decorators-legacy'
    ]
  }).code;
}

describe('add-react-displayname transform', () => {
  fs.readdirSync(fixturesDir).forEach(fixture => {
    const actual = transformFile(path.join(fixturesDir, fixture, 'input.js'));

    it(`transforms ${path.basename(fixture)}`, () => {
      expect(actual).toMatchSnapshot();
    });
  });
});
