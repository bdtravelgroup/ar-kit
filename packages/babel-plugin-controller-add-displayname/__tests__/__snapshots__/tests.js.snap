// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add-react-displayname transform transforms classComponents 1`] = `
"export let TestController = class TestController extends Controller {};
TestController.displayName = \\"TestController\\";"
`;
